CMakeLists.txt

```diff
@@ -0,0 +1,260 @@
+    libpd_wrapper/util/z_print_util.c
+    libpd_wrapper/util/z_print_util.h
+    libpd_wrapper/util/z_queued.c
+    libpd_wrapper/util/z_queued.h
+)
+
+source_group(pd FILES ${PD_SOURCES})
+source_group(pdextra FILES ${PD_EXTRA_SOURCES})
+source_group(libpd FILES ${LIBPD_SOURCES})
+source_group(libpdutils FILES ${LIBPD_UTILS_SOURCES})
+
+include_directories(libpd_wrapper)
+include_directories(pure-data/src)
+
+find_package(Threads REQUIRED)
+
+# create final list of source files
+set(SOURCE_FILES ${PD_SOURCES} ${LIBPD_SOURCES})
+if(PD_UTILS)
+    list(APPEND SOURCE_FILES ${LIBPD_UTILS_SOURCES})
+endif()
+if(PD_EXTRA)
+    list(APPEND SOURCE_FILES ${PD_EXTRA_SOURCES})
+endif()
+
+add_library(libpd_static STATIC ${SOURCE_FILES})
+add_library(libpd        SHARED ${SOURCE_FILES})
+
+# set the output library name for libpd depending on the settings
+set(LIBPD_OUTPUT_NAME     pd)
+if (WIN32)
+    set(LIBPD_OUTPUT_NAME libpd)
+endif()
+if (PD_MULTI)
+    set(LIBPD_OUTPUT_NAME ${LIBPD_OUTPUT_NAME}-multi)
+endif()
+if (WIN32)
+    set_target_properties(libpd_static PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME}-static)
+else()
+    set_target_properties(libpd_static PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME})
+endif()
+set_target_properties(libpd        PROPERTIES OUTPUT_NAME ${LIBPD_OUTPUT_NAME})
+
+# add appropriate compile definitions
+set(LIBPD_COMPILE_DEFINITIONS PD_INTERNAL)
+
+if(NOT MSVC)
+    list(APPEND LIBPD_COMPILE_DEFINITIONS HAVE_UNISTD_H=1)
+endif()
+
+if(PD_EXTRA)
+    list(APPEND LIBPD_COMPILE_DEFINITIONS LIBPD_EXTRA=1)
+endif()
+
+if(PD_MULTI)
+    list(APPEND LIBPD_COMPILE_DEFINITIONS PDINSTANCE=1 PDTHREADS=1)
+endif()
+
+if(NOT PD_LOCALE)
+    list(APPEND LIBPD_COMPILE_DEFINITIONS LIBPD_NO_NUMERIC=1)
+endif()
+
+if(UNIX)
+    list(APPEND LIBPD_COMPILE_DEFINITIONS HAVE_LIBDL)
+    target_link_libraries(libpd PUBLIC ${CMAKE_DL_LIBS})
+endif()
+
+target_compile_definitions(libpd_static PRIVATE ${LIBPD_COMPILE_DEFINITIONS})
+target_compile_definitions(libpd        PRIVATE ${LIBPD_COMPILE_DEFINITIONS})
+
+# link against necessary libraries
+if(MSVC)
+    target_link_libraries(libpd PUBLIC Ws2_32)
+    target_include_directories(libpd_static PUBLIC ${PTHREADS_INCLUDE_DIR})
+    target_include_directories(libpd        PUBLIC ${PTHREADS_INCLUDE_DIR})
+endif()
+target_link_libraries(libpd PUBLIC ${CMAKE_THREAD_LIBS_INIT})
+
+if(NOT MSVC AND NOT APPLE)
+    find_library(MATH_LIB m)
+    target_link_libraries(libpd PUBLIC ${MATH_LIB})
+endif()
+
+if(PD_BUILD_C_EXAMPLES)
+    macro(ADD_EXAMPLE name)
+        add_executable(${name} samples/c/${name}/${name}.c)
+        target_link_libraries(${name} PUBLIC libpd)
+        if(DEFINED CMAKE_DEBUG_POSTFIX)
+            set_target_properties(${name} PROPERTIES DEBUG_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+        endif()
+    endmacro(ADD_EXAMPLE)
+
+    ADD_EXAMPLE(pdtest)
+    if(PD_MULTI)
+        ADD_EXAMPLE(pdtest_multi)
+        if(NOT MSVC)
+            # uses gettimeofday()
```

Comment:
Instead of bypassing this example, it might be better to fix it so it builds on Windows. Possibly using https://stackoverflow.com/a/26085827

Reply:
I think it would be best to refactor the code not to use `gettimeofday()`. This should be easy to do as it is only used to sleep for some time.
I opened a separate issue for this: #246